examples/src/main/java/io/grpc/examples/errorhandling/ErrorHandlingClient.java

```diff
@@ -0,0 +1,220 @@
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.Uninterruptibles;
+
+import io.grpc.CallOptions;
+import io.grpc.ClientCall;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.Metadata;
+import io.grpc.Server;
+import io.grpc.ServerBuilder;
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+import io.grpc.examples.helloworld.GreeterGrpc;
+import io.grpc.examples.helloworld.GreeterGrpc.GreeterBlockingStub;
+import io.grpc.examples.helloworld.GreeterGrpc.GreeterFutureStub;
+import io.grpc.examples.helloworld.GreeterGrpc.GreeterStub;
+import io.grpc.examples.helloworld.HelloReply;
+import io.grpc.examples.helloworld.HelloRequest;
+import io.grpc.stub.StreamObserver;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.Nullable;
+
+/**
+ * Shows how to extract error information from a server response.
+ */
+public class ErrorHandlingClient {
+  public static void main(String [] args) throws Exception {
+    new ErrorHandlingClient().run();
+  }
+
+  private Server server;
+  private ManagedChannel channel;
+
+  void run() throws Exception {
+    server = ServerBuilder.forPort(0).addService(new GreeterGrpc.AbstractGreeter() {
+      @Override
+      public void sayHello(HelloRequest request, StreamObserver<HelloReply> responseObserver) {
+        responseObserver.onError(Status.INTERNAL
+            .withDescription("Eggplant Xerxes Crybaby Overbite Narwhal").asRuntimeException());
+      }
+    }).build().start();
+    channel =
+        ManagedChannelBuilder.forAddress("localhost", server.getPort()).usePlaintext(true).build();
+
+    blockingCall();
+    futureCallDirect();
+    futureCallCallback();
+    asyncCall();
+    advancedAsyncCall();
+
+    channel.shutdown();
+    server.shutdown();
+    channel.awaitTermination(1, TimeUnit.SECONDS);
+    server.awaitTermination();
+  }
+
+  void blockingCall() {
+    GreeterBlockingStub stub = GreeterGrpc.newBlockingStub(channel);
+    try {
+      stub.sayHello(HelloRequest.newBuilder().setName("Bart").build());
+    } catch (StatusRuntimeException e) {
+      Status status = e.getStatus();
+      Verify.verify(status.getCode() == Status.Code.INTERNAL);
+      Verify.verify(status.getDescription().contains("Eggplant"));
+      // Cause is not transmitted over the wire.
+    }
+  }
+
+  void futureCallDirect() {
+    GreeterFutureStub stub = GreeterGrpc.newFutureStub(channel);
+    ListenableFuture<HelloReply> response =
+        stub.sayHello(HelloRequest.newBuilder().setName("Lisa").build());
+
+    try {
+      response.get();
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException(e);
+    } catch (ExecutionException e) {
+      StatusRuntimeException sre = (StatusRuntimeException) e.getCause();
```

Comment:
If the exception is not `StatusRuntimeException`, this would throw a new exception and shadow the original one. I think it's better to use `Status.fromThrowable()` instead.